### ISSUE TITLE
Marshalling errors should provide more context

### ISSUE THREAD

**Issue by jbdeboer** [gocql/gocql#671]
Writing my program, I ran into the error:

```
2016/03/08 08:35:36 can not marshal int into timestamp
```

While accurate, I wasn't able to figure the accepted timestamp types until I dug through the source code.
A better error would be:

```
2016/03/08 08:35:36 can not marshal int into timestamp. Accepted types are Marshaler, int64, time.Time
```

**Comment by Zariel** [comment]
Yep this could be improved, but the marshaller should allow you to marshal an int into a timestamp column

**Comment by tengu-alt** [comment]
I'll handle it

**Comment by OleksiienkoMykyta** [comment]
@tengu-alt, looks like you forgot to add closes to your PR, it should be closed as resolved.
@joao-r-reis, could you help with it, please?

**Comment by joao-r-reis** [comment]
v2.0.0-rc1 with the fix for this has been released, it would be great if users could try it out and provide feedback on it before 2.0.0 GA
